src/main/java/com/exasol/validation/ProjectValidator.java

```diff
@@ -0,0 +1,112 @@
+        validateChanges(changes, version);
+    }
+
+    protected void validateChangelog(final String changelog, final String version) {
+        LOGGER.info("Validating changelog.md file.");
+        final String changelogContent = "[" + version + "](changes-" + version + ".md)";
+        if (!changelog.contains(changelogContent)) {
+            throw new IllegalStateException(
+                    "changelog.md file doesn't contain the following link, please add it to the file: "
+                            + changelogContent);
+        }
+    }
+
+    protected void validateChanges(final String changes, final String version) {
+        final String changesName = "changes-" + version + ".md";
+        LOGGER.info("Validating {} file.", changesName);
+        if (!changes.contains(version)) {
+            throw new IllegalStateException(changesName + " file doesn't contains a new version mentions.");
+        }
+        final String dateToday = new SimpleDateFormat("yyyy-MM-dd").format(Calendar.getInstance().getTime());
+        if (!changes.contains(dateToday)) {
+            throw new IllegalStateException("changes-" + version + ".md file doesn't contain release's date: "
+                    + dateToday + ". PLease, add or update the release date.");
+        }
+    }
+
+    /**
+     * Validate release requirements depending on the platform.
+     * 
+     * @param releasePlatform release platform
+     */
+    public void validatePlatform(final ReleasePlatform releasePlatform) {
+        if (releasePlatform == ReleasePlatform.GITHUB) {
+            validateGitHub();
+        } else {
+            throw new IllegalArgumentException(
+                    "Validation for release platform " + releasePlatform + " is not supported");
+        }
+    }
+
+    protected void validateGitHub() {
+        LOGGER.info("Validating github specific requirements.");
+        final Optional<String> latestReleaseTag = this.repository.getLatestReleaseVersion();
+        final String version = this.repository.getVersion();
+        validateVersion(version, latestReleaseTag);
+    }
+
+    protected void validateVersion(final String version, final Optional<String> latestReleaseTag) {
+        LOGGER.info("Validating a release version.");
+        if (latestReleaseTag.isPresent()) {
+            final Set<String> possibleVersions = getPossibleVersions(latestReleaseTag.get());
+            if (!possibleVersions.contains(version)) {
+                throw new IllegalStateException(
+                        "A new version doesn't fit the versioning rules. Possible versions for the release are: "
+                                + possibleVersions.toString());
+            }
+        } else {
+            final String[] versionParts = version.split("\\.");
+            if (versionParts.length != 3) {
+                throw new IllegalStateException(
+                        "The version has invalid format. The valid format is: <major>.<minor>.<fix>");
```

Comment:
Also please check that each part consists of only numbers. For this job a regular expression is better than a split.

Reply:
I wouldn't do that until we discuss and agree on the team that the tags contain only numbers. Because Muhammet, for example, uses something like 'v1.0.0' in his projects